### PR TITLE
Write file sync permissions

### PR DESCRIPTION
**Pull request by NewLunarFire**
Support file permissions in writeFileSync: Unix mode bits and append/truncate modes.

### CODE REVIEW

**Comment by ry** [comment]
Hmm - this seems to rely on some unix-only functionality 
```
error[E0433]: failed to resolve. Could not find `unix` in `os`
 --> ../../src\fs.rs:8:14
  |
8 | use std::os::unix::fs::PermissionsExt;
  |              ^^^^ Could not find `unix` in `os`
error[E0433]: failed to resolve. Use of undeclared type or module `PermissionsExt`
  --> ../../src\fs.rs:35:26
   |
35 |     file.set_permissions(PermissionsExt::from_mode(perm & 0o777))?;
   |                          ^^^^^^^^^^^^^^ Use of undeclared type or module `PermissionsExt`
warning: unused import: `std::os::unix::fs::PermissionsExt`
 --> ../../src\fs.rs:8:5
  |
8 | use std::os::unix::fs::PermissionsExt;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: #[warn(unused_imports)] on by default
error: aborting due to 2 previous errors
```
Is there a way to do this on windows too?

**Comment by NewLunarFire** [comment]
@ry fs.rs:L34-36, I added a cfg guard to set the mode bits only if we're in Unix. I don't know if I could completely prevent Rust from compiling those lines under Windows.

File: js/unit_tests.ts [hunk 1]
```diff
@@ -110,6 +110,33 @@ test(function makeTempDirSyncPerm() {
...
+
+testPerm({ write: true },async function writeFileSyncTruncate() {
+  const enc = new TextEncoder();
+  const dec = new TextDecoder();
+  const data1 = enc.encode("Hello");
+  const data2 = enc.encode("World");
+  const filename = "test.txt";
+
+  deno.writeFileSync(filename, data1);
+  deno.writeFileSync(filename, data2);
```
Comment (ry) [inline js/unit_tests.ts]: Could you use deno.makeTempDirSync() here?

File: js/unit_tests.ts [hunk 2]
```diff
@@ -110,6 +110,33 @@ test(function makeTempDirSyncPerm() {
...
+testPerm({ write: true },async function writeFileSyncAppend() {
+  const enc = new TextEncoder();
+  const dec = new TextDecoder();
+  const data1 = enc.encode("Hello");
+  const data2 = enc.encode(" World");
+  const filename = "test.txt";
+  const appendPerm = (0o666 | (1 << 31)); // Read-write + append
+
+  deno.writeFileSync(filename, data1);
+  deno.writeFileSync(filename, data2, appendPerm);
```
Comment (ry) [inline js/unit_tests.ts]: Could you use deno.makeTempDirSync() here?

**Comment by ry** [comment]
Sorry @NewLunarFire this one has gotten too far behind to land. We still haven't looked into how we're going to handle permission bits - but we will soon. I appreciate the work and apologize for not fighting to push this in.

**Pull request closed without merging**